src/solvers/dgsem_structured/dg_2d.jl

```diff
@@ -12,6 +12,9 @@ function rhs!(du, u, t,
   # Reset du
   @trixi_timeit timer() "reset ∂u/∂t" reset_du!(du, dg, cache)
 
+  # Apply waterheight and velocity cut off
+  apply_thresholds!(u, equations, dg, cache)
```

Comment:
You should not modify `u` inside the `rhs!` call. This violates assumptions of OrdinaryDiffEq.jl (and every other ODE solver library). Use a stage callback similar to the Zhang & Shu limiter.

Reply:
Ok, I did not know about this... @andrewwinters5000 had the idea to add the logic of `apply_thresholds!` after my implementation of the Zhang & She limiter. There is a special one needed for the SWE anyway because the last variable (bottom topography) must not be changed by the limiter. So this would not even create new methods.

Reply:
Yes, Sven and I discussed in the Slack that we could combine the Zhang & Shu limiter and his thresholding into a specialized version of the limiter. This strategy was born from the fact that the time integration method can only have a single stage limiter. So, the limiting + thresholding should be done in a single, specialzed routine that is unique for the Shallow Water situation.